python_modules/dagster/dagster/_core/definitions/unresolved_asset_job_definition.py

```diff
@@ -187,6 +190,32 @@ def resolve(
                 f'Error resolving selection for asset job "{self.name}": {e}'
             ) from e
 
+        # Require that all assets in the job have the same backfill policy
+        backfill_policies = {
+            job_asset_graph.get(k).backfill_policy for k in job_asset_graph.executable_asset_keys
+        }
+        if len(backfill_policies) > 1:
+            raise DagsterInvalidDefinitionError(
+                f"Asset job {self.name} materializes asset with varying BackfillPolicies. All assets"
+                " in a job must share the same BackfillPolicy."
+            )
+
+        # Error if a PartitionedConfig is defined and any target asset has a backfill policy that
+        # materializes anything other than a single partition per run. This is because
+        # PartitionedConfig is a function that maps single partition keys to run config, so it's
+        # behavior is undefined for multiple-partition runs.
```

Comment:
should we deprecate `PartitionedConfig`?

Reply:
The answer to this depends on the broader question of the future of job backfills and partitioned jobs.